src/masternodes/mn_checks.cpp

```diff
@@ -2692,21 +2700,40 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
                         return Res::Err("Cannot payback this amount of loan for %s, either payback full amount or less than this amount!", loanToken->symbol);
             }
 
-            res = mnview.SubMintedTokens(loanToken->creationTx, subLoan);
-            if (!res)
-                return res;
+            if (height < Params().GetConsensus().FortCanningHillHeight || kv.first != DCT_ID{0})
+            {
+                res = mnview.SubMintedTokens(loanToken->creationTx, subLoan);
+                if (!res)
+                    return res;
+            }
 
             CalculateOwnerRewards(obj.from);
-            // subtract loan amount first, interest is burning below
-            res = mnview.SubBalance(obj.from, CTokenAmount{kv.first, subLoan});
-            if (!res)
-                return res;
 
-            // burn interest Token->USD->DFI->burnAddress
+            // sub amount from balance and burn interest Token->USD->DFI->burnAddress
             if (subInterest)
             {
-                LogPrint(BCLog::LOAN, "CLoanTakeLoanMessage(): Swapping %s interest to DFI - %lld, height - %d\n", loanToken->symbol, subInterest, height);
-                res = SwapToDFIOverUSD(mnview, kv.first, subInterest, obj.from, consensus.burnAddress, height);
+                if (height < Params().GetConsensus().FortCanningHillHeight || kv.first != DCT_ID{0})
+                {
+                    // subtract loan amount first, interest is burning below
+                    res = mnview.SubBalance(obj.from, CTokenAmount{tokenId, subLoan});
+                    if (!res)
+                        return res;
+
+                    LogPrint(BCLog::LOAN, "CLoanTakeLoanMessage(): Swapping %s interest to DFI - %lld, height - %d\n", loanToken->symbol, subInterest, height);
+                    res = SwapToDFIOverUSD(mnview, tokenId, subInterest, obj.from, consensus.burnAddress, height);
+                }
+                else
+                {
+                    CAmount subLoanInDFI = mnview.GetAmountInCurrency(subLoan, {"DFI","USD"}, false, true, true);
+                    res = TransferTokenBalance(DCT_ID{0}, subLoanInDFI, obj.from, consensus.burnAddress);
```

Comment:
Just use `subBalance` defi amount, loaned tokens aren't burned.

Reply:
But DFI should be like burned. And we should introduce the tracking of it and that is still not done.
```The primary objective during burn is to keep a way to track just these burns in a dedicated manner. One option is to use a dedicated address to burn, so we have separate accountability, to allow for this economics be tweaked in the future. Any other options should also be acceptable if it helps the implementation easier.```
I think this is also meant for DFI token, but maybe I misunderstood it.

Reply:
`subBalance` is well enough here.

Reply:
As now loan and interest are burned combined in one action, I will leave it to transfer balance to burn address if you agree?